exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -592,11 +592,14 @@ public BatchGroup mergeAndSpill(LinkedList<BatchGroup> batchGroups) throws Schem
       }
       injector.injectChecked(context.getExecutionControls(), INTERRUPTION_WHILE_SPILLING, IOException.class);
       newGroup.closeOutputStream();
-    } catch (Exception e) {
+    } catch (Throwable e) {
       // we only need to cleanup newGroup if spill failed
-      AutoCloseables.close(e, newGroup);
+      try {
+        AutoCloseables.close(e, newGroup);
+      } catch (Throwable t) { /* close() may hit the same IO issue; just ignore */ }
```

Comment:
It looks like  close(Throwable t, AutoCloseable) suppresses the exception; did you get an exception during testing ?  Otherwise, you could remove this second try-catch.